From 5dd28d1aa5c5866425e48ae1151ea21a3d6d4d06 Mon Sep 17 00:00:00 2001
From: John Fredriksson <john.fredriksson@intel.com>
Date: Thu, 19 Nov 2015 18:11:33 +0100
Subject: [PATCH 05/28] Revert "atomisp: Switch of ISP if firmware loading
 fails"

This reverts commit aa8e181e3c64dac884aa4c7aff417641a8a3e868.

Conflicts:
	drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c

Change-Id: Iad365980cc73b98dc5866739922b8a01a9b0a8a8
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9279
Reviewed-on: https://android.intel.com:443/440491
---
 .../pci/atomisp2/atomisp_driver/atomisp_v4l2.c     | 24 ----------------------
 1 file changed, 24 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
index d6ea0b9..2988b96 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
@@ -1266,7 +1266,6 @@ static int atomisp_pci_probe(struct pci_dev *dev,
 	unsigned int start;
 	void __iomem *base;
 	int err, val;
-	u32 irq;
 
 	if (!dev) {
 		dev_err(&dev->dev, "atomisp: error device ptr\n");
@@ -1583,29 +1582,6 @@ enable_msi_fail:
 fw_validation_fail:
 	release_firmware(isp->firmware);
 load_fw_fail:
-	/*
-	 * Switch off ISP, as keeping it powered on would prevent
-	 * reaching S0ix states.
-	 *
-	 * The following lines have been copied from atomisp suspend path
-	 */
-
-	pci_read_config_dword(dev, PCI_INTERRUPT_CTRL, &irq);
-	irq = irq & 1 << INTR_IIR;
-	pci_write_config_dword(dev, PCI_INTERRUPT_CTRL, irq);
-
-	pci_read_config_dword(dev, PCI_INTERRUPT_CTRL, &irq);
-	irq &= ~(1 << INTR_IER);
-	pci_write_config_dword(dev, PCI_INTERRUPT_CTRL, irq);
-
-	atomisp_msi_irq_uninit(isp, dev);
-
-	atomisp_ospm_dphy_down(isp);
-	if (ATOMISP_INTERNAL_PM) {
-		if (atomisp_mrfld_power_down(isp))
-			dev_err(&dev->dev, "Failed to switch off ISP\n");
-	}
-
 	pci_dev_put(isp->pci_root);
 	return err;
 }
-- 
1.9.1

